### PR TITLE
Samba adjustments and cleanup

### PR DESCRIPTION
**Pull request by Omnione**
**_I affirm:_**
- [x] that I agree to Topaz Next's [Limited Contributor License Agreement](https://github.com/topaz-next/topaz/blob/release/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

Cleaned up the code a bit and adjusted a few things to make everything play a little nicer.
This should work with mobs using the ability also for when trust get to use sambas as an ability.

### CODE REVIEW

**Comment by zach2good** [comment]
I hate this section of code _so so much_, god bless you for trying to fix it!

Content seems fine, but can you please style it:
https://github.com/topaz-next/topaz/blob/release/.github/CONTRIBUTING.md#breakbeforebraces-allman

**Comment by Omnione** [comment]
I blame who ever wrote it initially, i hate the 
if (x == 5) {
}

**Comment by Omnione** [comment]
Im gonna go ahead and style the whole function

File: src/map/utils/battleutils.cpp [hunk 1]
```diff
@@ -939,11 +942,20 @@ namespace battleutils
         if (PAttacker->getMod(Mod::ENSPELL) > 0 &&
             (PAttacker->getMod(Mod::ENSPELL_CHANCE) == 0 || PAttacker->getMod(Mod::ENSPELL_CHANCE) > tpzrand::GetRandomNumber(100)))
         {
-            static SUBEFFECT enspell_subeffects[8] = {
-                SUBEFFECT_FIRE_DAMAGE,      SUBEFFECT_ICE_DAMAGE,   SUBEFFECT_WIND_DAMAGE,  SUBEFFECT_EARTH_DAMAGE,
-                SUBEFFECT_LIGHTNING_DAMAGE, SUBEFFECT_WATER_DAMAGE, SUBEFFECT_LIGHT_DAMAGE, SUBEFFECT_DARKNESS_DAMAGE,
+            static SUBEFFECT enspell_subeffects[8] =
```
Comment (ghost) [inline src/map/utils/battleutils.cpp:949]: Is there a reason we're defining this here?  This looks like duplicated code from enum SUBEFFECT.  Totally a result of formatting and not additional code though.

**Comment by zach2good** [comment]
Oh, I linked you the wrong section! This is what I meant to show you:
https://github.com/topaz-next/topaz/blob/release/.github/CONTRIBUTING.md#braces-around-statements

File: src/map/utils/battleutils.cpp (hunk 1 above)
Comment (Omnione) [reply src/map/utils/battleutils.cpp:949]: No idea, way before my time

**Comment by Omnione** [comment]
Why you no like my braces change Linux

File: src/map/utils/battleutils.cpp (hunk 1 above)
Comment (zach2good) [reply src/map/utils/battleutils.cpp:949]: All element's _should_ be aligned globally now, so tables like this shouldn't be needed anymore. But that's outside of the scope of this PR 👍

**Review by zach2good — APPROVED** [review]
Good luck! I've had to revert 2 sets of changes in this area in the past, I hope you fare better! 🤞

**Pull request merged**